debugoverlay-ext-netstats/src/main/java/com/ms_square/debugoverlay_ext_netstats/NetStatsDataModule.java

```diff
@@ -0,0 +1,76 @@
+package com.ms_square.debugoverlay_ext_netstats;
+
+import android.net.TrafficStats;
+import android.os.Handler;
+import android.os.Looper;
+
+import com.ms_square.debugoverlay.modules.BaseDataModule;
+
+import java.util.Locale;
+
+public class NetStatsDataModule extends BaseDataModule<String> {
+
+    private static final String TAG = NetStatsDataModule.class.getSimpleName();
+
+    private static final String HEADER = "Received: %8s/s\nSent: %12s/s";
+
+    private Handler handler = new Handler(Looper.getMainLooper());
+
+    private final int interval;
+    private final int uid;
+
+    private double mPreviousReceived = 0;
+    private double mPreviousSent = 0;
+    private double mReceived;
+    private double mSent;
+
+    public NetStatsDataModule(int interval, int uid) {
+        this.interval = interval;
+        this.uid = uid;
+    }
+
+    @Override
+    protected String getLatestData() {
+        return String.format(
+                Locale.ENGLISH, HEADER, bytesToPrettyString(mReceived), bytesToPrettyString(mSent));
+    }
+
+    @Override
+    public void start() {
+        handler.post(networkStatisticsQueryRunnable);
+    }
+
+    @Override
+    public void stop() {
+        handler.removeCallbacks(networkStatisticsQueryRunnable);
+    }
+
+    private final Runnable networkStatisticsQueryRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            double seconds = interval * 0.001;
```

Comment:
you don't need to convert this into second at every run because interval is final and you always get the same value.